packages/block-library/src/page-list/convert-to-links-modal.js

```diff
@@ -107,12 +51,13 @@ export default function ConvertToLinksModal( { onClose, clientId } ) {
 				<Button
 					variant="primary"
 					disabled={ ! pagesFinished }
-					onClick={ convertSelectedBlockToNavigationLinks( {
-						pages,
-						replaceBlock,
-						clientId,
-						createBlock: create,
-					} ) }
+					onClick={ () => {
+						const navigationLinks =
+							convertToNavigationLinks( pages );
+
+						// Replace the Page List block with the Navigation Links.
+						replaceBlock( clientId, navigationLinks );
```

Comment:
This new call is needed but that has the advantage of making the function simpler and more flexible.

Reply:
And easier to test!